lib/core/src/display/shape/text/text_field/render/assignment.rs

```diff
@@ -230,6 +227,7 @@ impl<'a,'b,'c> GlyphLinesAssignmentUpdate<'a,'b,'c> {
     fn new_assignment(&mut self) -> RangeInclusive<usize> {
         let visible_lines         = self.visible_lines_range();
         let assigned_lines        = &self.assignment.assigned_lines;
+        let max_line_id           = self.content.lines.len().saturating_sub(1);
```

Comment:
what does "max line id" mean ? Is there a maximum id a line could have? And what is `saturating_sub(1)` ?

Reply:
> what does "max line id" mean ? Is there a maximum id a line could have_ 
Yes, indeed.
> And what is `saturating_sub(1)`?
A little google: https://doc.rust-lang.org/std/primitive.usize.html?search=#method.saturating_sub